weblate/api/views.py

```diff
@@ -404,10 +404,36 @@ def new_template(self, request, **kwargs):
 
         return self.download_file(obj.get_new_base_filename(), 'application/binary')
 
-    @action(detail=True, methods=['get'])
+    @action(detail=True, methods=['get', 'post'])
     def translations(self, request, **kwargs):
         obj = self.get_object()
 
+        if request.method == 'POST':
+            if not request.user.has_perm('translation.add', obj):
+                self.permission_denied(request, message='Can not create translation')
+
+            if 'language_code' not in request.data:
+                raise ParseError("Missing 'language_code' parameter")
+
+            language_code = request.data['language_code']
+
+            if not Language.objects.filter(code__in=(language_code,)):
+                raise Http404("No language code '%s' found!" % language_code)
+
+            language = Language.objects.get(code=language_code)
```

Comment:
I'd rather use try/except here instead of checking for existence and then retrieving the object again from the database.

```suggestion
            try:
                language = Language.objects.get(code=language_code)
            except Language.DoesNotExist:
                raise Http404("No language code '%s' found!" % language_code)
```

Reply:
Done. I rebased onto non-broken upstream master and use `try/except` here.